### ISSUE TITLE
More examples needed

### ISSUE THREAD

**Issue by renich** [mvz/gir_ffi-gtk#12]
Hello,

IMHO, the examples are awesome; but too few. 

I've requested collaboration around google+; specifically, in the ruby, gnome and gtk+ communities.

Having a full text editor or some sample app that shows how to use it would be awesome, if you ask me.

**Comment by aanno** [comment]
For a hobby I need some gtk3 support for rubinius and the rubinius people suggested to have a try on gir_ffi-gtk (instead of ruby-gnome2). 

As I trying to get started, I forked in order to include the 'sample' directory from [ruby-gnome2](https://github.com/ruby-gnome2/ruby-gnome2) in a [sample_of_gtk3](https://github.com/aanno/gir_ffi-gtk/tree/sample_of_gtk3) branch.

It was startling easy to 'convert' a few of the examples in the misc directory to gir_ffi-gtk (and the example even WORK with rubinius, which is *gorgeous* compared to ruby-gnome2 on rubinius).

That said, I still feel that the project is in its infancy because nothing from the more complex stuff works. The major blockers seems to be:

* No inheritance from the gtk3 widgets, at least not the (easy) way that is done in ruby-gnome2. I've sawn the remark on [derived types](https://github.com/mvz/gir_ffi/blob/master/TODO.md#derived-types) at gir_ffi but for me the present state looks like a blocker. Perhaps I could give a helping hand with this but at the moment I'm not sure how archive better inheritance support.
* No support for Gdk::Pixbuf (at least not in this namespace). This is critical, as most more complex examples use some sort of images and/or icons, hence Pixbuf is *always* involved. I guess you could get some support with plain gir_ffi but I haven't fiddle it out yet. Some advice would be welcome.

**Comment by aanno** [comment]
After more many more trying, I was able to get some of the Pixbuf stuff running (sample/misc/aboutdialog.rb). It seems to be as easy as <code><pre>
GirFFI.setup :GdkPixbuf
a.logo = GdkPixbuf::Pixbuf.new_from_file("gnome-logo-icon.png")
</pre></code>
I'm starting to see some elegance in the gir stuff of Gtk for foreign language bindings :smirk:.

Perhaps a (command line) tool that could dump info about the Repository objects would come in handy, especially for novice (like me). Ideally, the tool should be able to emit (pseudo) ruby code to see what attributes and methods there are, how many arguments (and of which type) they expect etc. I even think this could compensate for the missing examples and/or documentation.

I also had a try on sample/misc/treeview.rb. However, I found it extremely hard to even *construct* a TreeStore, and hence didn't arrived at anything. Even with the [Gtk documentation](https://developer.gnome.org/gtk3/stable/GtkTreeStore.html), I have no idea what are the (ruby mapped) types of arguments to pass to the constructor/new method.

**Comment by aanno** [comment]
It turns out that there is already a tool chain that could read from GIR and could extract some documentation.
<code><pre>
g-ir-generate /usr/lib/girepository-1.0/cairo-1.0.typelib >cairo.xml
g-ir-doc-tool -o cairo-folder cairo.xml
yelp-build html cairo-folder
</code></pre>
There are some more details about this at http://mathematicalcoffee.blogspot.ch/2012/09/developing-gnome-shell-extensions_6.html .
However, sometimes the GIR information is not appropriate. Cairo of the example above is such a case.

**Comment by mvz** [comment]
Yes, ideally this would be the tool to use. Best if it can produce Ruby-specific output.
